doc/guides/env_vars.rst

```diff
@@ -0,0 +1,169 @@
+you make in RapidEE will be available whenever you open a new terminal window.
+
+.. _env_vars_zephyrrc:
+
+Option 3: Using ``zephyrrc`` files
+----------------------------------
+
+Choose this option if you don't want to make the variable's setting available
+to all of your terminals, but still want to save the value for loading into
+your environment when you are using Zephyr.
+
+**macOS and Linux**:
+
+Create a file named :file:`~/.zephyrrc` if it doesn't exist, then add this line
+to it:
+
+.. code-block:: console
+
+   export MY_VARIABLE=foo
+
+To get this value back into your current terminal environment, **you must run**
+``source zephyr-env.sh`` from the main ``zephyr`` repository. Among other
+things, this script sources :file:`~/.zephyrrc`.
+
+The value will be lost if you close the window, etc.; run ``source
+zephyr-env.sh`` again to get it back.
+
+**Windows**:
+
+Add the line ``set MY_VARIABLE=foo`` to the file
+:file:`%userprofile%\\zephyrrc.cmd` using a text editor such as Notepad to save
+the value.
+
+To get this value back into your current terminal environment, **you must run**
+``zephyr-env.cmd`` in a ``cmd.exe`` window after changing directory to the main
+``zephyr`` repository.  Among other things, this script runs
+:file:`%userprofile%\\zephyrrc.cmd`.
+
+The value will be lost if you close the window, etc.; run ``zephyr-env.cmd``
+again to get it back.
+
+Option 4: Using Zephyr Build Configuration CMake package
+--------------------------------------------------------
+
+Choose this option if you want to make those variable settings shared among all
+users of your project.
+
+Using a :ref:`cmake_build_config_package` allows you to commit the shared
+settings into the repository, so that all users can share them.
+
+It also removes the need for running ``source zephyr-env.sh`` or
+``zephyr-env.cmd`` when opening a new terminal.
+
+.. _zephyr-env:
+
+Zephyr Environment Scripts
+**************************
+
+You can use the zephyr repository scripts ``zephyr-env.sh`` (for macOS and
+Linux) and ``zephyr-env.cmd`` (for Windows) to load Zephyr-specific settings
+into your current terminal's environment. To do so, run this command from the
+zephyr repository::
+
+  # macOS and Linux
+  source zephyr-env.sh
+
+  # Windows
+  zephyr-env.cmd
+
+These scripts:
+
+- set :envvar:`ZEPHYR_BASE` (see below) to the location of the zephyr
+  repository
+- adds some Zephyr-specific locations (such as zephyr's :file:`scripts`
+  directory) to your :envvar:`PATH` environment variable
+- loads any settings from the ``zephyrrc`` files described above in
+  :ref:`env_vars_zephyrrc`.
+
+You can thus use them any time you need any of these settings.
+
+.. _env_vars_important:
+
+Important Environment Variables
+*******************************
+
+Here are some important environment variables and what they contain. This is
+not a comprehensive index to the environment variables which affect Zephyr's
+behavior.
```

Comment:
So if we cannot create a comprehensive list or link to one, how will users of Zephyr ever be able to discover what settings that can be controlled using environment variables ?

Reply:
> So if we cannot create a comprehensive list or link to one, how will users of Zephyr ever be able to discover what settings that can be controlled using environment variables ?

As usual, they will have to read the source code and ask questions from the developers. I hate this as much as you do, but it is important to at least document reality.

Reply:
> > So if we cannot create a comprehensive list or link to one, how will users of Zephyr ever be able to discover what settings that can be controlled using environment variables ?
> 
> As usual, they will have to read the source code and ask questions from the developers. I hate this as much as you do, but it is important to at least document reality.

Ok, I created a complete list of all variables Zephyr supports as environment settings.
```
ZEPHYR_BASE
BOARD
SHIELD
CONF_FILE

WEST_DIR
ZEPHYR_EXTRA_MODULES
ZEPHYR_BOARD_ALIASES (Not supported as regular CMake variables, points to file, wrongly documented)

Toolchains:
TOOLCHAIN_ROOT
ZEPHYR_SDK_INSTALL_DIR
ZEPHYR_TOOLCHAIN_VARIANT (see list of possible values)
  - espressif
  - armclang
  - xtensa
  - llvm
  - arcmwdt
  - xtools
  - host
<toolchain_upper>_TOOLCHAIN_PATH

CROSS-COMPILE
ONEAPI_ROOT (alternative to ONEAPI_TOOLCHAIN_PATH)
XCC_USE_CLANG (Not supported as regular CMake variable, not documented)

CLANG_ROOT_DIR (deprecated, use LLVM_TOOLCHAIN_PATH)
GCCARMEMB_TOOLCHAIN_PATH (deprecated, use GNUARMEMB_TOOLCHAIN_PATH)


Emulators:
QEMU_BIN_PATH (has a CMake possibility of `QEMU=<path-to-qemu>`)
OVMF_FD_PATH (Not supported as regular CMake variables)
QEMU_EXTRA_FLAGS (not documented)
ARMFVP_BIN_PATH (has a CMake possibility of `ARMFVP=<path-to-armfvp>`)
ARMFVP_BL1_FILE
ARMFVP_FIP_FILE

Board specific env (Not supported as regular CMake variables)
EVERGLADES_SPI_GEN
EVERGLADES_SPI_CFG
MEC172X_SPI_GEN
MEC172X_SPI_CFG

Babble sim (Not supported as regular CMake variables):
BSIM_COMPONENTS_PATH
BSIM_OUT_PATH
```

I would say we should list the top-ones, and the important toolchain related:
`ZEPHYR_TOOLCHAIN_VARIANT`: (same descr as now)
`<toolchain_upper>_TOOLCHAIN_PATH: The path to the toolchain specified in `ZEPHYR_TOOLCHAIN_VARIANT`.
If `ZEPHYR_TOOLCHAIN_VARIANT=llvm` then this path is `LLVM_TOOLCHAIN_PATH`.

**Note**: `DTC_OVERLAY_FILE` is not being picked up from environment.

Reply:
we can then refer to board and emulator environment variables in more generic manner.

Something like:
> Emulators and boards may depend on additional programs.
> The build system will try to locate those programs automatically, but may rely on additional CMake or environment variables to do so.
> Please consult the emulator and/or board specific documentation for more information.

That way we can provide a complete list of general build settings to the reader, while still point to the fact that emulators and boards may have more settings.

At least that would make me happier.

Reply:
see also: https://github.com/zephyrproject-rtos/zephyr/issues/40389

Reply:
> Ok, I created a complete list of all variables Zephyr supports as environment settings.

OK, but can we deal with that in a separate PR? The scope of this PR is just to resurrect the environment variables page.

Reply:
Accepted.